wandb/integration/ultralytics/callback.py

```diff
@@ -0,0 +1,367 @@
+import copy
+from datetime import datetime
+from typing import Callable, Dict, Optional, Union
+
+try:
+    import dill as pickle
+except ImportError:
+    import pickle
+
+import torch
+from tqdm.auto import tqdm
+
+try:
+    from ultralytics.yolo.engine.model import TASK_MAP, YOLO
+    from ultralytics.yolo.utils import RANK, __version__
+    from ultralytics.yolo.utils.torch_utils import de_parallel
+    from ultralytics.yolo.v8.pose.train import PoseTrainer
+    from ultralytics.yolo.v8.pose.val import PoseValidator
+    from ultralytics.yolo.v8.pose.predict import PosePredictor
+    from ultralytics.yolo.v8.detect.train import DetectionTrainer
+    from ultralytics.yolo.v8.detect.val import DetectionValidator
+    from ultralytics.yolo.v8.detect.predict import DetectionPredictor
+    from ultralytics.yolo.v8.classify.train import ClassificationTrainer
+    from ultralytics.yolo.v8.classify.val import ClassificationValidator
+    from ultralytics.yolo.v8.classify.predict import ClassificationPredictor
+except ImportError as e:
+    print(e)
+
+import wandb
+from wandb.integration.ultralytics.pose_utils import (
+    plot_pose_predictions,
+    plot_pose_validation_results,
+)
+from wandb.integration.ultralytics.bbox_utils import (
+    plot_predictions,
+    plot_validation_results,
+)
+from wandb.integration.ultralytics.classification_utils import (
+    plot_classification_predictions,
+    plot_classification_validation_results,
+)
+from wandb.sdk.lib import telemetry
+
+
+class WandBUltralyticsCallback:
```

Comment:
Change name to `WandbUltralyticsCallback` also, Does the callback work with YOLOv5 and YOLOv7 models ? if not, we should probably rename it to `WandbYOLOv8Callback`